docs_src/design/adr/0019-Service-List.md

```diff
@@ -0,0 +1,87 @@
+
+The current list of needs include:
+
+- Security bootstrapping service requires a list of services in order to know which services need a file token for Vault access and to add shared secrets to Vault.
+- System management agent service requires a list of services in order to know which services it can provide status, configuration and metrics for client requestors.  The SMA also needs to know which services to include in any start/stop/restart operations.
+- The user interface needs to know which services, specifically device and application services, are available in order to provide interaction with those services, and potentially to display some metrics on operational services.
+- The CLI needs to know which services are operational (for some interaction) and which device services and application services to include in start/stop operations.
+
+In the future, there may be other internal or 3rd party needs to what services comprise an EdgeX deployment.
+
+### Current Situation
+
+Currently, the list of services is haphazardly managed by the needing service.
+
+- The security bootstrapping services are being built to use an environment variable to provide it with a list of services at startup.
+- The user interface and CLI do not deal with service lists today and assume all services are operational (with known service host and port defined in configuration).
+- The SMA has a static/hardcoded list built into the codebase.  Clearly the least desirable means of dealing with the list of services as this not only requires a restart of the service, but also code changes and redeployment.
+
+### Requirements
+
+1. It is desired that the service list be addressed in a consistent manner.  It is further desired that there be one single, authoritative source of the service list information for all EdgeX services and 3rd party requestors.  While the source of the service list may change under certain runtime circumstances (for example – whether operating with or without security), once EdgeX is running the authoritative source remains the same for all services.
+
+2. Any changes to the list of services must adhere to ACID transactional properties.  That is, any change to the list of services must be atomic, consistent, isolated and durable with regard to any operations on the service list.  In other words, no two services should ever be able to get a different list of services when making a request of EdgeX for the list of services.
+
+3. EdgeX services may be added or removed (either intentionally or unintentionally due to failures) all the time.  The list of services should reflect, as best it can, the current list of EdgeX services – that is be dynamic and not a static list at the start/bootstrap of the EdgeX instance.  In particular, new device services and application services are likely to be added to the EdgeX instance after the system is started.  The list should eventually reflect the additions.
```

Comment:
The phrases "reflect, as best it can" and "should eventually reflect" are consistency violations, as defined in a previous comment.

Reply:
agree this made it seem non-ACID and subject to eventual consistency.  I took out this language.